mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/settings.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming
+
+object SessionFlowSettings {
+
+  /**
+   * Factory method for Scala.
+   */
+  def apply(maxPacketSize: Int): SessionFlowSettings =
+    new SessionFlowSettings(maxPacketSize)
+
+  /**
+   * Java API
+   *
+   * Factory method for Java.
+   */
+  def create(maxPacketSize: Int): SessionFlowSettings =
+    SessionFlowSettings(maxPacketSize)
+}
+
+final class SessionFlowSettings private (val maxPacketSize: Int) {
```

Comment:
No validation of maxPacketSize?

Reply:
I'm unsure what it could be validated to...

Reply:
Actually, I guess it cannot exceed 0xffff and it must be greater or equal to zero... lemme add something

Reply:
Done